.circleci/config.yml

```diff
@@ -26,3 +26,41 @@ jobs:
       - run: stack exec codegen-exe
       - run: stack test --jobs 2
       - run: stack exec ffi-test
+  cabal-build:
+    docker:
+      - image: "ubuntu:18.04"
+    environment:
+      LD_LIBRARY_PATH: /root/project/deps/libtorch/lib:/root/project/deps/mklml/lib
+      PATH: /opt/ghc/bin:/bin:/usr/bin:/usr/local/bin:/sbin:/usr/sbin
+    steps:
+      - run: echo $LD_LIBRARY_PATH
+      - run: apt update -qq && apt install -y cmake curl wget unzip git libtinfo-dev python3 python3-yaml
+      - run: apt -y --allow-downgrades --allow-remove-essential --allow-change-held-packages install locales software-properties-common apt-transport-https
+      - run: add-apt-repository ppa:hvr/ghc
+      - run: apt-get update -qq && apt-get -y --allow-downgrades --allow-remove-essential --allow-change-held-packages install build-essential zlib1g-dev liblapack-dev libblas-dev ghc-8.6.4 cabal-install-head devscripts debhelper python3-pip
+      - run: update-alternatives --install /usr/bin/python python /usr/bin/python3 1
+      - checkout
+      - run: git submodule init && git submodule update
+      - run: gcc --version
+      - run: cabal --version
+      - run: cd deps/ ; ./get-deps.sh USE_BINARY_FOR_CI
+      - run: cd libtorch-test ; make
+      - run: ./setup-cabal.sh
+      - run: cabal new-update
+      - run: cabal new-install hspec-discover
+      - run:
+          name: cabal new-build all
+          command: cabal new-build all --jobs=2
+          no_output_timeout: 15m
+      - run: cabal new-test ffi --jobs=2
```

Comment:
`doctest` does not work on `cabal` and can not find packages, because `cabal` does not set `GHC_PACKAGE_PATH`.